docs/index.md

```diff
@@ -128,6 +128,12 @@ This can be problematic when run in `__init__.py` files, which often contain
 the `--ignore-init-modules` flag, which will filter away any passed
 `__init__.py` files before processing.
 
+There may be import statements that are being used dynamically, 
+to autoimport it would look like those are not being used but actually they may have some usages.
+
+In such cases where you want to retain the unused imports across any file, you can use the `--keep-unused-imports` flag, 
```

Comment:
I'd add a note that if there are not that many cases, using  `# noqa: autoimport` should be favoured over using the flag.